feat(security): add Spring Security, roles, Basic Auth, Swagger security and Liquibase seeding

vDmytriv01 opened the pull request:
- add SecurityConfig: BCryptPasswordEncoder, stateless sessions, httpBasic, permitAll for /auth/register, /swagger-ui/**, /v3/api-docs/**, /error
- enable method security; protect /books/** with @PreAuthorize
- add CustomUserDetailsService (load user by email)
- add Role/RoleName, RoleRepository; add Set<Role> roles to User
- registration flow: encode password, assign USER role by default
- OpenAPI: add basicAuth SecurityScheme; mark /books/** with @SecurityRequirement
- Liquibase: create users/roles/users_roles/books tables; seed roles (USER/ADMIN); seed users (admin@mail.com, user@mail.com) with bcrypt(1234);

vDmytriv01 committed:
feat(security): add Spring Security, roles, Basic Auth, Swagger security and Liquibase seeding

- add SecurityConfig: BCryptPasswordEncoder, stateless sessions, httpBasic,
  permitAll for /auth/register, /swagger-ui/**, /v3/api-docs/**, /error
- enable method security; protect /books/** with @PreAuthorize
- add CustomUserDetailsService (load user by email)
- add Role/RoleName, RoleRepository; add Set<Role> roles to User
- registration flow: encode password, assign USER role by default
- OpenAPI: add basicAuth SecurityScheme; mark /books/** with @SecurityRequirement
- Liquibase: create users/roles/users_roles/books tables; seed roles (USER/ADMIN);
  seed users (admin@mail.com, user@mail.com) with bcrypt(1234);

Elena-Bruyako commented on src/main/java/com/vdmytriv/bookstoreapp/controller/AuthenticationController.java:
```diff
@@ -16,14 +16,14 @@
 
     @ResponseStatus(HttpStatus.CREATED)
-    @PostMapping("/registration")
+    @PostMapping("/register")
```
according to the task description, path should be "registration"

Elena-Bruyako commented on src/main/java/com/vdmytriv/bookstoreapp/model/Role.java:
```diff
@@ -0,0 +1,29 @@
+import jakarta.persistence.Table;
+import org.springframework.security.core.GrantedAuthority;
+
+@Entity
```
```suggestion
@Entity
@Getter
@Setter
```

Elena-Bruyako commented on src/main/java/com/vdmytriv/bookstoreapp/model/User.java:
```diff
@@ -38,4 +46,22 @@ public class User {
     @Column(nullable = false)
     private boolean isDeleted = false;
+
+    @ManyToMany(fetch = FetchType.EAGER)
```
don't use EAGER
```suggestion
    @ManyToMany
```

Elena-Bruyako commented on src/main/java/com/vdmytriv/bookstoreapp/model/User.java:
```diff
@@ -38,4 +46,22 @@ public class User {
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id")
+    )
+    private Set<Role> roles;
```
```suggestion
    private Set<Role> roles = new HashSet<>();
```

Elena-Bruyako commented on src/main/java/com/vdmytriv/bookstoreapp/model/User.java:
```diff
@@ -38,4 +46,22 @@ public class User {
+    @Override
+    public String getUsername() {
+        return email;
+    }
```
override other methods

Elena-Bruyako commented on src/main/java/com/vdmytriv/bookstoreapp/model/User.java:
```diff
@@ -38,4 +46,22 @@ public class User {
+    public String getUsername() {
+        return email;
+    }
 }
```
add soft-delete approach

Elena-Bruyako commented on src/main/java/com/vdmytriv/bookstoreapp/repository/user/UserRepository.java:
```diff
@@ -1,8 +1,11 @@
 public interface UserRepository extends JpaRepository<User, Long> {
     boolean existsByEmail(String email);
+
+    Optional<User> findByEmail(String email);
```
add EntityGraph with roles

Elena-Bruyako commented on src/main/java/com/vdmytriv/bookstoreapp/service/user/UserServiceImpl.java:
```diff
@@ -4,24 +4,35 @@
         }
-        User user = userMapper.toModel(requestDto);
+        Role userRole = roleRepository.findByName(RoleName.USER)
+                .orElseThrow(() -> new RuntimeException("Default role USER not found"));
```
use RoleName.USER in message

Elena-Bruyako commented on src/main/java/com/vdmytriv/bookstoreapp/service/user/UserServiceImpl.java:
```diff
@@ -4,24 +4,35 @@
 import org.springframework.stereotype.Service;
 
 @RequiredArgsConstructor
 @Service
```
```suggestion
@Service
@Transactional
```

Elena-Bruyako reviewed (changes requested):
Good job! Let's improve your solution

vDmytriv01 committed:
refactor(security): apply reviewer suggestions for user and role management

- AuthenticationController: fix endpoint mapping to /registration
- Role: clean duplicate @Entity annotation
- User: remove eager fetch, initialize roles with HashSet, override UserDetails methods
- UserRepository: add EntityGraph to fetch roles with user
- UserServiceImpl: add @Transactional and improve exception message with RoleName.USER
